docs/developer-guide/api.rst

```diff
@@ -326,3 +326,67 @@ If a CREATE, DROP, or TERMINATE statement returns a command status with state QU
         "status": "SUCCESS",
         "message":"Stream created and running"
       }
+
+
+Terminate A Cluster
+-------------------
+
+If you don't need your KSQL cluster anymore, you can terminate the cluster and clean up the resources using this endpoint. When a server receives a ``TERMINATE CLUSTER`` request at ``/ksql/terminate`` endpoint, it writes a ``TERMINATE CLUSTER`` command into the command topic.
+Note that ``TERMINATE CLUSTER`` request can only be sent via the ``/ksql/terminate`` endpoint and you cannot send it via the CLI.
+When each server in the cluster reads a ``TERMINATE CLUSTER`` command, it takes the following steps:
+
+-Sets the KSQL engine mode to ``NOT ACCEPTING NEW STATEMENTS`` so no new statement will be passed to the engine for execution.
+-Terminates all of persistent and transient queries in the engine along with the required clean up for each query.
+-Deletes all Kafka topics associated with the streams/tables in the KSQL metastore.
+-Deletes the command topic for the cluster.
+
+**Example request**
+
+   .. code:: http
+
+      POST /ksql/terminate HTTP/1.1
+      Accept: application/vnd.ksql.v1+json
+      Content-Type: application/vnd.ksql.v1+json
+
+      {
+        "streamsProperties": {}
+      }
+
+You can customize the clean up process if you don't want to delete all of the Kafka topics associated with the streams/tables in the metastore:
+
+**Provide a List of Sources to Keep**
+If you want to keep a set of topics associated with streams/tables in the metastore, provide the list of stream/table names in your request along with the type of this list.
+This will clean up all topics belonging to the Streams/Tables in the metastore except for the ones that belong to the streams/tables that are in the ``SOURCES_LIST`` list.
+Note that you need to indicate the type of the list as ``KEEP`` by setting the value of ``SOURCES_LIST_TYPE`` parameter to ``KEEP``.
+The following example will keep the Kafka topics for streams/tables named ``FOO`` and ``BAR``.
+
+   .. code:: http
+
+      POST /ksql/terminate HTTP/1.1
+      Accept: application/vnd.ksql.v1+json
+      Content-Type: application/vnd.ksql.v1+json
+
+      {
+        "streamsProperties": {
+          "SOURCES_LIST": ["FOO", "BAR"],
```

Comment:
Rather than needing `SOURCE_LIST` and `SOURCE_LIST_TYPE`, why not just have 

```
"streamsProperties": {
    "SOURCES_LIST": ["FOO", "BAR"],
    "SOURCES_LIST_TYPE": "KEEP"
}
```

Why not just have:

```
"streamsProperties": {
    "keepList": ["FOO", "BAR"]
}
```

For the white list, or

```
"streamsProperties": {
    "deleteList": ["FOO", "BAR"]
}
```

for the black list. (Failing if both are provided).